code/game/gamemodes/miniantags/revenant/revenant.dm

```diff
@@ -121,9 +121,22 @@
 	remove_from_all_data_huds()
 
 	spawn(5)
-		if(src.mind)
-			src.mind.wipe_memory()
-			src << 'sound/effects/ghost.ogg'
+		if(mind)
+			giveObjectivesandGoals()
+		if(!giveSpells())
+			message_admins("Revenant was created but has no mind. Trying again in ten seconds.")
+			addtimer(CALLBACK(src, .proc/giveSpellsObjectivesordelete), 10 SECONDS)
```

Comment:
Also this being in New() seems a poor choice functionally. So if an admin spawns a revenant and after 10 seconds it has no mind it's automatically deleted?

Reply:
Yeah, I haven't put much thought into why this was a bad idea in the first place. I'll see how I can make this better.